src/components/viewSections/LicenseAgreements.js

```diff
@@ -67,9 +67,10 @@ export default class LicenseAgreements extends React.Component {
   }
 
   static getDerivedStateFromProps(props, state) {
-    const { license: { linkedAgreements } } = props;
+    const { license: { id, linkedAgreements } } = props;
 
     if (
+      (id !== state.groupedLinkedAgreements.remoteId) ||
```

Comment:
@CalamityC  This will work but I think its not the ideal way to solve the issue here. There is a larger issue here where the whole _ViewLicenseRoute_ isn't remounting when you navigate from one license to the other. This causes stripes-connect to inject the stale data from the previously fetched license and that data is trickling down into the nested component and causing an issue. 

We need to fix this issue at the top level index file where we are rendering the component in conjuncion with the `resourceShouldRefresh` option from stripes-connect.

There are a few different ways to achieve this, but one way could be replacing [this line](https://github.com/folio-org/ui-licenses/blob/master/src/index.js#L133) with 

```<Route path={`${path}/:id`} exact render={(props) => (
                    <ViewLicenseRoute 
                      key={props.match.params.id} 
                      {...props} 
                      />
                    )} 
                  />```

and also passing a `resourceShouldRefresh: true` param  [here](https://github.com/folio-org/ui-licenses/blob/master/src/routes/ViewLicenseRoute.js#L51)

They key prop would make sure that the component remounts when the `id` of the license changes. The `resourceShouldRefresh: true` would ensure stripes-connect resets the resource (linkedAgreements) when the component unmounts so that we dont get the stale results back. Let me know if that works?